### PR TITLE
Migrate to cassandra driver 4  - server/blob/blob-cassandra

### PR DESCRIPTION
**Pull request by vttranlina**
TODO: Some unit test throw error: Query timed out after PT2S

### CODE REVIEW

File: server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBucketDAO.java [hunk 1]
```diff
@@ -74,60 +73,64 @@ public CassandraBucketDAO(BlobId.Factory blobIdFactory, Session session) {
...
-            .where(eq(BUCKET, bindMarker(BUCKET)))
-            .and(eq(ID, bindMarker(ID)))
-            .and(eq(BucketBlobParts.CHUNK_NUMBER, bindMarker(BucketBlobParts.CHUNK_NUMBER))));
+    private PreparedStatement prepareSelectPart(CqlSession session) {
+        return session.prepare(selectFrom(BucketBlobParts.TABLE_NAME)
+                .all()
+             .whereColumn(BUCKET).isEqualTo(bindMarker(BUCKET))
+             .whereColumn(ID).isEqualTo(bindMarker(ID))
+             .whereColumn(BucketBlobParts.CHUNK_NUMBER).isEqualTo(bindMarker(BucketBlobParts.CHUNK_NUMBER))
+                .build());
```
Comment (chibenwa) [inline server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBucketDAO.java]: Indent looks weird...

**Review by chibenwa — COMMENTED** [review]
Don't you mind if I ask you to fix the tests before I merge this?

**Comment by vttranlina** [comment]
> Don't you mind if I ask you to fix the tests before I merge this?

No problem, 

// I tried to fix the test that throw an error, but it looks like taking more time than I thought.  
If this module blocks another module in the compile phase, we can merge it first.  (anyway ci still fails on the apache repository)

**Comment by chibenwa** [comment]
BTW which test is failing?

**Comment by chibenwa** [comment]
If it is `CassandraMailRepositoryWithFakeImplementationsTest`

Then:

 - queries to be instrumented no longer should end with `;`
 - names should be lowered cased

**Comment by vttranlina** [comment]
> BTW which test is failing?

`org.apache.james.blob.cassandra.cache.CachedBlobStoreTest#shouldCacheWhenReadWithDefaultBucket`

**Pull request merged**